hax/hax/ha/events.py

```diff
@@ -145,3 +152,78 @@ def ack(self):
            that message using ack method
         """
         self.consumer.ack()
+
+
+class EventProducer():
+    """
+    Module responsible for publishing the event to message bus
+    """
+
+    ADMIN_ID = "hare_admin"
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        """
+        Static method to fetch the current instance.
+        Performs initialization related to common database and message bus
+        """
+        if not EventProducer.__instance:
+            EventProducer()
+        return EventProducer.__instance
+
+    def __init__(self):
+        logging.debug('Inside EventProducer')
+
+        if EventProducer.__instance is None:
+            EventProducer.__instance = self
+        else:
+            raise Exception("EventProducer is singleton class,"
+                            " use EventProducer.get_instance().")
+
+        message_type = FAULT_TOLERANCE_KEYS.HARE_HA_MESSAGE_TYPE.value
+        self._register_producer(message_type)
+
+        self.producer = MessageProducer(producer_id=HEALTH_EVENT_SOURCES.HARE,
```

Comment:
So, in future if we need to post another message type do we need to created another producer instance? If so, then this design is not supporting having multiple producers per EventProducer instances or do we create multiple EventProducer instances?

Reply:
Util's MessageProducer class works on single message_type only.
So if we need to support multiple message_type then we can do it using multiple instances of 'EventProducer' by taking message_type as input while creating instances.

Reply:
Utils MessageProducer works on single message_type only.
So if we need to support multiple message_type then we can create multiple EventProducer instances by taking message_type(or entity based on which multiple message_types are created) as input during instance creation.
For e.g. if there are multiple message type 1 for each type of resource such as cluster, node, disk the we can take resource type as input and then create producer accordingly.